File: java/ankit_kanyal/devnest_problems/phase_1/ContainerWithMostWater.java
// link- https://devsnest.in/algo-challenges/container-with-most-water

static int solve(int n, int[] arr){
//CODE HERE 

int low=0,high=n-1;
int maxArea=0,curArea=0;

while(low<high){
    curArea=Math.min(arr[low],arr[high])*(high-low);
    maxArea=Math.max(maxArea,curArea);
    if(arr[low]<arr[high]) low++;
    else high--;
}
return maxArea;
}

File: Python/chandu/Container_With_Most_Water.py
def solve(n, arr):
    # CODE HERE
    # using two pointers....
    l, r = 0, n-1
    kat = 0
    while l<r:
        kat = max(kat, min(arr[l], arr[r])*(r-l))
        if arr[l]<arr[r]:
            l += 1
        else:
            r -= 1
    
    return kat
